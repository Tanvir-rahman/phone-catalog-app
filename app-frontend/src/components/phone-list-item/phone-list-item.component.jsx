import React from 'react';
import { withRouter } from 'react-router-dom';

import './phone-list-item.styles.scss';

export const PhoneListItemComponent = ({ id, title, price, imageurl, history }) => {
  return (
    <div onClick={() => history.push(`/phones/${id}`)}>
      {
        <div className="phone">
          <div className="phone-image">
            <img 
              src={require('@assets/images/' + imageurl)}
              alt={title}
            />
          </div>
          <div className="phone-info">
            <h5 className="info-title">{title}</h5>
            <h6 className="info-price">${price}</h6>
          </div>
        </div>
      }
    </div>
  );
}

export default withRouter(PhoneListItemComponent);