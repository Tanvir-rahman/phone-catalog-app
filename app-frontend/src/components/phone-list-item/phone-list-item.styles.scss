.phone {
  margin: 2%;
  padding: 10px;
  display: flex;
  flex-grow: 1;
  flex-basis: 16%;
  flex-direction: column;
  border: 1px solid #ccc;
  border-radius: 5px;
  text-align: center;
  cursor: pointer;

  .phone-image img {
    max-width: 100%;
  }
  .phone-info {
    margin-top: auto;
    h5 {
      font-size: 1.2rem;
      font-weight: bold;
      margin-bottom: 0.5rem;
    }
    h6 {
      font-size: 1rem;
      font-weight: normal;
      margin-top: 0;
    }
  }
}

@media (max-width: 920px) {
  .phone {
    flex: 1 21%;
  }
}

@media (max-width: 600px) {
  .phone {
    flex: 1 46%;
  }
}