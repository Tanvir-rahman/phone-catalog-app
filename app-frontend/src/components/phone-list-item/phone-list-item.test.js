import React from 'react';
import { shallow } from 'enzyme';

import { PhoneListItemComponent } from './phone-list-item.component';

describe('PhoneListItem component', () => {
  let wrapper;
  const mockId = 'iphone-11'
  const mockTitle = 'Iphone 11';
  const mockPrice = '100';
  const mockImageUrl = 'iphone-11.png';

  beforeEach(() => {
    const mockProps = {
      id: mockId,
      title: mockTitle,
      price: mockPrice,
      imageurl: mockImageUrl
    };

    wrapper = shallow(<PhoneListItemComponent {...mockProps} />);
  });

  it('should render PhoneListItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title prop in Name Containerr', () => {
    expect(wrapper.find('.info-title').text()).toBe(mockTitle);
  });

  it('should render price prop in Price Container', () => {
    const price =wrapper.find('.info-price').text();
    expect(price).toBe(`$${mockPrice}`);
  });
});
