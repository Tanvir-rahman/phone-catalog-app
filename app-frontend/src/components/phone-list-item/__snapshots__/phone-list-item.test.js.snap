// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneListItem component should render PhoneListItem component 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="phone"
  >
    <div
      className="phone-image"
    >
      <img
        alt="Iphone 11"
        src="iphone-11.png"
      />
    </div>
    <div
      className="phone-info"
    >
      <h5
        className="info-title"
      >
        Iphone 11
      </h5>
      <h6
        className="info-price"
      >
        $
        100
      </h6>
    </div>
  </div>
</div>
`;
